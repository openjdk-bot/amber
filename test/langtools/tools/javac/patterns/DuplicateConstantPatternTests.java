/*
 * @test /nodynamiccopyright/
 * @summary Checking for duplicate constant patterns test
 * @compile/fail/ref=DuplicateConstantPatternTests.out -XDrawDiagnostics DuplicateConstantPatternTests.java
 */

public class DuplicateConstantPatternTests {

    public static void main(String[] args) {
        int i = 47;
        switch (i) {
            case 45: throw new AssertionError("broken");
            case 45: throw new AssertionError("broken");
        }

        switch (i) {
            case 45: throw new AssertionError("broken");
            case 44+1: throw new AssertionError("broken");
        }

        switch (i) {
            case 45: throw new AssertionError("broken");
            case 46-1: throw new AssertionError("broken");
            default: System.out.println("Default");
        }

        Object o = "hello";
        switch (o) {
            case null: throw new AssertionError("broken");
            case null: throw new AssertionError("broken");
        }
    }

}
