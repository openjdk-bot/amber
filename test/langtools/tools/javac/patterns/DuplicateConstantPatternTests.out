DuplicateConstantPatternTests.java:13:13: compiler.err.duplicate.case.label
DuplicateConstantPatternTests.java:18:13: compiler.err.duplicate.case.label
DuplicateConstantPatternTests.java:23:13: compiler.err.duplicate.case.label
DuplicateConstantPatternTests.java:30:13: compiler.err.duplicate.case.label
4 errors